src/common/hooks/useZeroApprove.ts

```diff
@@ -4,18 +4,65 @@ import { useApproveCallback } from './useApproveCallback'
 import { Currency, CurrencyAmount } from '@uniswap/sdk-core'
 import { useSetAtom } from 'jotai'
 import { zeroApprovalState } from '../state/useZeroApprovalState'
+import { useSafeApiKit } from 'api/gnosisSafe/hooks/useSafeApiKit'
+import { SafeMultisigTransactionResponse } from '@safe-global/safe-core-sdk-types'
+import { useIsActiveWallet } from 'legacy/hooks/useIsActiveWallet'
+import { walletConnectConnection } from 'modules/wallet/web3-react/connection/walletConnect'
+import { useIsSafeWallet } from 'modules/wallet'
+
+function wait(ms = 1000) {
+  return new Promise((resolve) => {
+    setTimeout(resolve, ms)
+  })
+}
+
+async function poll<T>(
+  fn: () => Promise<T | null>,
+  fnCondition: (result: T | null) => boolean,
+  ms: number
+): Promise<T | null> {
+  let result = await fn()
+
+  while (fnCondition(result)) {
+    await wait(ms)
+    result = await fn()
+  }
+
+  return result
+}
```

Comment:
this looks pretty generic to me, maybe a more central place? like a common util?

Reply:
Looking at the implentation, it should be called `pollUntil`?

Reply:
Done